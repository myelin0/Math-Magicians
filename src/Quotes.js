import React, { useState, useEffect, useRef } from 'react';

const Quote = () => {
  const [quotes, setQuotes] = useState([]);
  const colors = ['#9e4200', '#421c00', '#628500', '#009025', '#391306', '#aa1945', '#3B0918', '#b8390e'];
  const textRef = useRef();
  const getQuotes = () => {
    fetch('https://type.fit/api/quotes')
      .then((res) => res.json())
      .then((data) => {
        const randomNumber = Math.floor(Math.random() * data.length);
        setQuotes(data[randomNumber]);
      });
  };

  useEffect(() => {
    getQuotes();
  }, []);

  useEffect(() => {
    textRef.current.style.color = colors[Math.floor(Math.random() * 8)];
  }, [quotes]);

  return (
    <div className="quote">
      <h2>Get motivated with new quote</h2>
      <p className="quote-body" ref={textRef}>
        {quotes.text}
      </p>
      <strong>
        {' '}
        ~
        {quotes.author || 'anonyme'}
      </strong>
      <div className="btnContainer">
        <button
          type="button"
          className="btn btn-warning"
          onClick={getQuotes}
        >
          Get Quote
        </button>
        <a
          href={`http://twitter.com/intent/tweet?text=${quotes.text}`}
          target="_blank"
          rel="noopener noreferrer"
          className="btn btn-warning"
        >
          tweet
          {/* <FaTwitter style={{ color: '#1DA1F2', fontSize: '16px', marginLeft: '5px' }} /> */}
          <i style={{ color: '#1DA1F2', fontSize: '16px', marginLeft: '5px' }} className="fa fa-twitter" />
        </a>
      </div>
    </div>
  );
};

export default Quote;
