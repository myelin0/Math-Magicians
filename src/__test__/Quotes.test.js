import React from 'react';
import renderer from 'react-test-renderer';
import Quotes from '../Quotes';

it('renders correctly', () => {
  const tree = renderer.create(<Quotes />).toJSON();
  expect(tree).toMatchSnapshot();
});
