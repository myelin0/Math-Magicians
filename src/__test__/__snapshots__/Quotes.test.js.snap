// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="quote"
>
  <h2>
    Get motivated with new quote
  </h2>
  <p
    className="quote-body"
  />
  <strong>
     
    ~
    anonyme
  </strong>
  <div
    className="btnContainer"
  >
    <button
      className="btn btn-warning"
      onClick={[Function]}
      type="button"
    >
      Get Quote
    </button>
    <a
      className="btn btn-warning"
      href="http://twitter.com/intent/tweet?text=undefined"
      rel="noopener noreferrer"
      target="_blank"
    >
      tweet
      <i
        className="fa fa-twitter"
        style={
          Object {
            "color": "#1DA1F2",
            "fontSize": "16px",
            "marginLeft": "5px",
          }
        }
      />
    </a>
  </div>
</div>
`;
